src/github.com/matrix-org/dendrite/clientapi/sync/syncserver.go

```diff
@@ -38,6 +45,28 @@ func (s *Server) Start() error {
 }
 
 func (s *Server) onMessage(msg *sarama.ConsumerMessage) error {
-	log.WithField("key", string(msg.Key)).WithField("val", string(msg.Value)).Info("Recv")
+	// Parse out the event JSON
+	var output api.OutputRoomEvent
+	if err := json.Unmarshal(msg.Value, &output); err != nil {
+		// If the message was invalid, log it and move on to the next message in the stream
+		log.WithError(err).Errorf("roomserver output log: message parse failure")
+		return nil
+	}
+
+	ev, err := gomatrixserverlib.NewEventFromTrustedJSON(output.Event, false)
+	if err != nil {
+		log.WithError(err).Errorf("roomserver output log: event parse failure")
+		return nil
+	}
+	log.WithFields(log.Fields{
+		"event_id": ev.EventID(),
+		"room_id":  ev.RoomID(),
+	}).Info("received event from roomserver")
+
+	if err := s.db.WriteEvent(&ev, output.AddsStateEventIDs, output.RemovesStateEventIDs); err != nil {
+		log.WithError(err).Errorf("roomserver output log: write event failure")
```

Comment:
I feel like we need to worry more about this. Not sure what we can do about it though.